bumble/host.py

```diff
@@ -563,13 +585,19 @@ def on_hci_connection_complete_event(self, event):
                 f'### BR/EDR CONNECTION: [0x{event.connection_handle:04X}] '
                 f'{event.bd_addr}'
             )
+            # For connection initiator, role change may be issued if the remote device tries to switch role.
+            # For connection resopnder, role change is always issued.
+            if role := self.pending_connection_roles.get(event.bd_addr):
```

Comment:
This should probably test for `is not None`, because if `pending_connection_roles[event.bd_addr]` is `BT_CENTRAL_ROLE`, (which has value 0), this will not delete the entry.

Reply:
```
            if event.bd_addr in self.pending_connection_roles:
                role = self.pending_connection_roles[event.bd_addr]
                del self.pending_connection_roles[event.bd_addr]
```